WordPress/wordpress-develop
# Update theme block patterns to current shape of used blocks

Trac ticket: https://core.trac.wordpress.org/ticket/53617


There have been many enhancements and added features regarding block patterns which require their parsing in browser's idle time, during editor's loading.

If we have deprecated versions of blocks in the patterns, they produce some info log messages in console. While this is not something that causes any problem, it can be a bit overwhelming if someone uses the dev tools.

## Some details about the changes
The diff is kind of rough in some themes:

2021 - some of the deprecations :

- Image blocks with align left, right or center needed a div wrapper and the appropriate classes
- A Group block with the extra div wp-block-group__inner-container
- Column blocks with a number in width like <!-- wp:column {"width":80}. The current shape has width as a string, so it should be <!-- wp:column {"width":"80%"}

Most deprecations for the other themes where about the extra div in `Group` block and the `borderRadius` attribute in `Button` block.


## Testing instructions

1. Verify that every pattern's design is not affected and can be inserted with no problems as before
2. Verify that there are no logs produced from deprecated block conversions

## Notes
There is still one conversion coming from core patterns fetched by the pattern/directory. If you disable core patterns you'll see no conversion log messages.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
